docs/source/python/data.rst

```diff
@@ -230,6 +230,28 @@ like lists:
    nested_arr = pa.array([[], None, [1, 2], [None, 1]])
    print(nested_arr.type)
 
+ListView arrays
+~~~~~~~~~~~
+
+``pyarrow.array`` can create an alternate list type called ListView:
+
+.. ipython:: python
+
+   nested_arr = pa.array([[], None, [1, 2], [None, 1]], type=pa.list_view(pa.int64()))
+   print(nested_arr.type)
+
+ListView arrays have a different set of buffers than List arrays. The ListView array
+has both an offsets and sizes buffer, while a List array only has an offsets buffer.
+This allows for ListView arrays to specify out-of-order offsets:
```

Comment:
❤️

Reply:
Is it easy/possible to add a link to the spec from here?

Reply:
Sure. We could add 
```
:meth:`~pyarrow.ListViewArray`
```
to reference to the PyArrow API docs or add a reference to the existing reference object in the format/Columnar.rst section
```
:ref:`listview-layout`
```

Reply:
I can add that! Great idea

Reply:
https://github.com/apache/arrow/pull/40679